File: python/주식가격.py
def solution(prices):
    answer,s=[0 for i in range(len(prices))],[]
    for idx,val in enumerate(prices):
        if len(s)==0:
            s.append([val,idx]);
        else:
            if s[-1][0] > val:
                while len(s)!=0 and s[-1][0] >val:
                    answer[s[-1][1]]=idx-s[-1][1]
                    s.pop()
            s.append([val,idx])
    for i in s:
        answer[i[1]]=len(prices)-i[1]-1
    return answer

File: java/주식가격.java
import java.util.*;
class Solution {
    Stack<pair> stack = new Stack<>();
    public class pair{
        public int a,b;
        pair(int a, int b){
            this.a = a;
            this.b = b;
        }
    }
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        
        for(int i=0;i<prices.length;i++){
            while(!stack.empty()&&stack.peek().b>prices[i]){
                answer[stack.peek().a]=i-stack.peek().a;
                stack.pop();
            }
            stack.push(new pair(i,prices[i]));
        }
        while(!stack.empty()){
                answer[stack.peek().a]=prices.length-stack.peek().a-1;
                stack.pop();
        }
        return answer;
    }
}